tests/bluetooth/bsim_bt/bsim_test_l2cap/src/main_l2cap_ecred.c

```diff
@@ -0,0 +1,602 @@
+{
+	struct bt_l2cap_server *server;
+
+	server = get_free_server();
+	if (!server) {
+		FAIL("Failed to get free server");
+		return;
+	}
+
+	server->accept = accept;
+	server->psm = 0;
+
+	if (bt_l2cap_server_register(server) < 0) {
+		FAIL("Failed to get free server");
+		return;
+	}
+
+	LOG_DBG("L2CAP server registered, PSM:0x%X", server->psm);
+}
+
+static void connected(struct bt_conn *conn, uint8_t conn_err)
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+
+	if (conn_err) {
+		FAIL("Failed to connect to %s (%u)", addr, conn_err);
+		bt_conn_unref(default_conn);
+		default_conn = NULL;
+		return;
+	}
+
+	default_conn = bt_conn_ref(conn);
+	LOG_DBG("%s", addr);
+
+	SET_FLAG(is_connected);
+}
+
+static void disconnected(struct bt_conn *conn, uint8_t reason)
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+
+	LOG_DBG("%s (reason 0x%02x)", addr, reason);
+
+	if (default_conn != conn) {
+		FAIL("Conn mismatch disconnect %s %s)", default_conn, conn);
+		return;
+	}
+
+	bt_conn_unref(default_conn);
+	default_conn = NULL;
+	UNSET_FLAG(is_connected);
+}
+
+BT_CONN_CB_DEFINE(conn_callbacks) = {
+	.connected = connected,
+	.disconnected = disconnected,
+};
+
+static void send_sdu(int iteration, int chan_idx, int bytes)
+{
+	struct bt_l2cap_chan *chan = &channels[chan_idx].le.chan;
+	struct net_buf *buf;
+
+	/* First 4 bytes in sent payload is iteration count */
+	sys_put_le32(iteration, channels[chan_idx].payload);
+
+	if (channels[chan_idx].buf != 0) {
+		FAIL("Buf should have been dealocated by now");
+		return;
+	}
+
+	if (chan_idx == 0) {
+		buf = net_buf_alloc(&tx_data_pool_0, K_NO_WAIT);
+	} else {
+		buf = net_buf_alloc(&tx_data_pool_1, K_NO_WAIT);
+	}
+
+	if (buf == NULL) {
+		FAIL("Failed to get buff on ch %i, iteration %i should never happen", chan_idx,
+		     chan_idx);
+	}
+
+	channels[chan_idx].buf = buf;
+	net_buf_reserve(buf, BT_L2CAP_CHAN_SEND_RESERVE);
+	net_buf_add_mem(buf, channels[chan_idx].payload, bytes);
+
+	LOG_DBG("bt_l2cap_chan_sending ch: %i bytes: %i iteration: %i", chan_idx, bytes, iteration);
+	const int ret = bt_l2cap_chan_send(chan, buf);
+
+	LOG_DBG("bt_l2cap_chan_send returned: %i", ret);
+
+	if (ret < 0) {
+		FAIL("Error: send failed error: %i", ret);
+		channels[chan_idx].buf = 0;
+		net_buf_unref(buf);
+	}
+}
+
+void send_sdu_chan_worker(struct k_work *item)
+{
+	const struct channel *ch = CONTAINER_OF(item, struct channel, work);
+
+	send_sdu(ch->iteration, ch->chan_id, ch->bytes_to_send);
+}
+
+static void send_sdu_concurrently(void)
+{
+	for (int i = 0; i < SDU_SEND_COUNT; i++) {
+		for (int k = 0; k < L2CAP_CHANNELS; k++) {
+			channels[k].iteration = i;
+			/* Assign the right msg to the right channel */
+			channels[k].bytes_to_send = (k == LONG_MSG_CHAN_IDX) ? LONG_MSG : SHORT_MSG;
+			const int err = k_work_submit_to_queue(&channels[k].work_queue,
+									&channels[k].work);
+
+			if (err < 0) {
```

Comment:
Looks like the alignment of the split line for `k_work_submit_to_queue()` isn't quite right here either. Also, unless dealing with variable declarations at the top of a block, the convention within the Bluetooth subsystem is not to have an empty line between the assignment to a variable (like `err`) and testing for that variable's value.

Reply:
And as for variable declarations at the top of a block, at leat to me it looks easier to read like this:
```c
{
	int err;

	err = some_func();
	if (err) {
		...
	}
```

Reply:
It is quite tedious fo fix this kind of endless style issues. I would propose to make as many as automatic checks as necessary and what is not detected by the automatic checks to be a bit more tolerant with it. Otherwise we will have endless PRs with nitpicking comments on them which will definitely be counterproductive.